Modules/Filtering/BiasCorrection/include/itkN4BiasFieldCorrectionImageFilter.hxx

```diff
@@ -66,11 +67,49 @@ N4BiasFieldCorrectionImageFilter<TInputImage, TMaskImage, TOutputImage>
   this->m_MaximumNumberOfIterations.Fill( 50 );
 }
 
+
+template<typename TInputImage, typename TMaskImage, typename TOutputImage>
+void
+N4BiasFieldCorrectionImageFilter<TInputImage, TMaskImage, TOutputImage>
+::VerifyPreconditions() ITKv5_CONST
+{
+  Superclass::VerifyPreconditions();
+
+  const InputImageType *const inputImage = this->GetInput();
+
+  if (inputImage == nullptr)
+  {
+    itkExceptionMacro("Failed precondition: Input image should not be null");
+  }
+
+  const typename InputImageType::SizeType inputImageSize = inputImage->GetBufferedRegion().GetSize();
+
+  const MaskImageType *const maskImage = GetMaskImage();
+
+  if ((maskImage != nullptr) && (maskImage->GetBufferedRegion().GetSize() != inputImageSize))
+  {
+    itkExceptionMacro(
+      "Failed precondition: If a mask image is specified, its size should be equal to the input image size");
```

Comment:
The documentation for [VerifyPreconditions](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/Common/include/itkProcessObject.h#L666-L677) says it is called before `UpdateOutputInformation` is called. The is a step in the ITK pipeline before the filters GenerateData step is done. So, the LargestPossibleRegion is not even updated at this point in the pipeline.

Reply:
@blowekamp Can you please explain? The `maskImage` is an input of the filter. So it should be checked before GenerateData. OK?